device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/auth/SignatureHelper.java

```diff
@@ -80,13 +80,9 @@
             hMacSha256.init(secretKey);
             encryptedSig = hMacSha256.doFinal(sig);
         }
-        catch (NoSuchAlgorithmException e)
+        catch (NoSuchAlgorithmException | InvalidKeyException e)
         {
-            // should never happen, since the algorithm is hard-coded.
-        }
-        catch (InvalidKeyException e)
```

Comment:
can we say for sure that InvalidKeyException won't happen? should we throw incase it does? (since algorithm is hard-coded, NoSuchAlgorithmException might not happen; but it seems to me that InvalidKeyException might still be thrown).

Reply:
This is kind of a public API, so I'd rather not start throwing this exception. I can certainly add a log statement to help with debugging though